### PR TITLE
check dependencies present in registry on package push

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

check dependencies present in registry on package push

## Fixes

https://github.com/fetchai/agents-aea/issues/1859

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: aea/cli/registry/push.py [hunk 1]
```diff
@@ -143,11 +145,20 @@ def push_item(ctx: Context, item_type: str, item_id: PublicId) -> None:
     }
 
     # dependencies
+    dependecies: List[Tuple[str, PublicId]] = []
```
Comment (DavidMinarsch) [inline aea/cli/registry/push.py:148]:
```suggestion
    dependencies: List[Tuple[str, PublicId]] = []
```

File: aea/cli/registry/push.py [hunk 2]
```diff
@@ -143,11 +145,20 @@ def push_item(ctx: Context, item_type: str, item_id: PublicId) -> None:
...
     for key in [CONNECTIONS, CONTRACTS, PROTOCOLS, SKILLS]:
-        deps_list = item_config.get(key)
+        deps_list = item_config.get(key, [])
         if deps_list:
             data.update({key: deps_list})
+        for dep in deps_list:
+            dependecies.append((ITEM_TYPE_PLURAL_TO_TYPE[key], PublicId.from_str(dep)))
```
Comment (DavidMinarsch) [inline aea/cli/registry/push.py:154]:
```suggestion
            dependencies.append((ITEM_TYPE_PLURAL_TO_TYPE[key], PublicId.from_str(dep)))
```

File: aea/cli/registry/utils.py [hunk 3]
```diff
@@ -311,3 +322,21 @@ def get_latest_version_available_in_registry(
         )
 
     return latest_item_public_id
+
+
+def list_missing_packages(
+    packages: List[Tuple[str, PublicId]]
+) -> List[Tuple[str, PublicId]]:
+    """Get list of packages do not present in registry."""
```
Comment (DavidMinarsch) [inline aea/cli/registry/utils.py:330]:
```suggestion
    """Get list of packages not currently present in registry."""
```

File: aea/cli/registry/push.py [hunk 4]
```diff
@@ -143,11 +145,20 @@ def push_item(ctx: Context, item_type: str, item_id: PublicId) -> None:
...
 
+    missing_dependencies = list_missing_packages(dependecies)
+
+    if missing_dependencies:
+        for package_type, package_id in missing_dependencies:
+            click.echo(f"Error: Can not find {package_type} {package_id} in registry!")
+        raise click.ClickException("Found missed dependencies! Push canceled!")
```
Comment (DavidMinarsch) [inline aea/cli/registry/push.py:161]:
```suggestion
        raise click.ClickException("Found missing dependencies! Push canceled!")
```

File: aea/cli/registry/push.py [hunk 5]
```diff
@@ -143,11 +145,20 @@ def push_item(ctx: Context, item_type: str, item_id: PublicId) -> None:
...
         if deps_list:
             data.update({key: deps_list})
+        for dep in deps_list:
+            dependencies.append((ITEM_TYPE_PLURAL_TO_TYPE[key], PublicId.from_str(dep)))
 
+    missing_dependencies = list_missing_packages(dependecies)
+
+    if missing_dependencies:
+        for package_type, package_id in missing_dependencies:
+            click.echo(f"Error: Can not find {package_type} {package_id} in registry!")
```
Comment (marcofavorito) [inline aea/cli/registry/push.py:160]:
```suggestion
            click.echo(f"Error: Cannot find {package_type} {package_id} in registry!")
```

**Review by marcofavorito — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by 5A11 — APPROVED** [review]
lgtm

**Comment by 5A11** [comment]
Please ensure CI passes.

**Pull request merged**